libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/JwtHelper.java

```diff
@@ -8,38 +8,42 @@
 import com.nimbusds.jose.JWSHeader;
 import com.nimbusds.jose.JWSSigner;
 import com.nimbusds.jose.crypto.impl.ECDSA;
+import com.nimbusds.jose.jwk.ECKey;
 import com.nimbusds.jose.util.Base64URL;
 import com.nimbusds.jwt.JWTClaimsSet;
 import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.ipv.core.library.service.ConfigService;
 
+import java.text.ParseException;
 import java.util.Map;
 
-public class JwtHelper {
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.SIGNING_KEY_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.SIGNING_KEY_JWK;
+import static uk.gov.di.ipv.core.library.config.CoreFeatureFlag.KID_JAR_HEADER;
 
-    private static final ObjectMapper mapper = new ObjectMapper();
+public class JwtHelper {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOGGER = LogManager.getLogger();
 
     private JwtHelper() {}
 
-    public static <T> SignedJWT createSignedJwtFromObject(T claimInput, JWSSigner signer)
-            throws JOSEException {
-        JWSHeader jwsHeader = generateHeader();
+    public static <T> SignedJWT createSignedJwtFromObject(
```

Comment:
I wonder if we should change this signature to take a `JWTClaimsSet` instead, and get rid of `generateClaims`...

Looking at the other usage (for the private key JWT), we'd be better served getting rid of our `ClientAuthClaims` object and using the built-in nimbus classes. E.g.:
```
            var clientAuthClaims = new JWTClaimsSet.Builder()
                    .issuer(criConfig.getClientId())
                    .subject(criConfig.getClientId())
                    .audience(criConfig.getComponentId())
                    .expirationTime(...)
                    .jwtID(secureTokenHelper.generate())
                    .build();
```
IMO this is clearer and safer than using our custom class and serialising ourselves! And it means you can use a shared method for the signing.

Appreciate it's nothing really to do with your change, so happy to put into a different PR if you prefer.

Reply:
Yep, that makes sense. I'll fix it up here as I'm in that headspace.